src/coreclr/src/System.Private.CoreLib/ILLinkTrim.xml

```diff
@@ -9,5 +9,13 @@
     <type fullname="System.Diagnostics.Tracing.EventPipe*" />
     <!-- The private Event methods are accessed by private reflection in the base EventSource class. -->
     <type fullname="System.Diagnostics.Tracing.NativeRuntimeEventSource" />
+    <!-- Accessed via native code. -->
+    <type fullname="System.Runtime.InteropServices.ComTypes.IEnumerable" />
+    <type fullname="System.Runtime.InteropServices.ComTypes.IEnumerator" />
+    <type fullname="System.Runtime.InteropServices.CustomMarshalers.*" />
+    <!-- Workaround for https://github.com/mono/linker/issues/378 -->
+    <type fullname="System.Runtime.InteropServices.IDispatch" />
+    <type fullname="System.Runtime.InteropServices.IMarshal" />
```

Comment:
Technically, this type does exist on mono when `'$(TargetsWindows)' == 'true'`. 

@jkotas @AaronRobinsonMSFT - do you think this is the right decision? Or should I keep it in the "shared" ILLInkTrim since that is where the type exists?

Reply:
The `IMarshal` should only be used for the `StandardOleMarshalObject` which is for COM only. Since mono doesn't support COM at all, I think any usage of `StandardOleMarshalObject` on the mono runtime is technically unsupported. Therefore keeping it in coreclr is probably fine even though usage in a mono scenario would be broken. @jkotas may have another opinion but I think the risk is minimal for user impact.

Reply:
The support for Mono on Windows overall is incomplete in multiple ways. We are not shipping Mono as supported product (in .NET 5 at least).

Mono CoreLib does not ever define `FeatureComInterop` and so it gets stubs for other COM interop parts too: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems#L1605

A potential alternative would be to put this under `FeatureComInterop` instead of Windows/Unix, but I do not think it would be material improvement. There are number of other places in libraries that depend on COM interop to work on Windows.